ple/games/tetris_engine.py

```diff
@@ -0,0 +1,518 @@
+                     '.....',
+                     '.....',
+                     '.....']]
+
+O_SHAPE_TEMPLATE = [['.OO..',
+                     '.OO..',
+                     '.....',
+                     '.....',
+                     '.....']]
+
+J_SHAPE_TEMPLATE = [['.O...',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..OO.',
+                     '..O..',
+                     '..O..',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '...O.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '.OO..',
+                     '.....',
+                     '.....']]
+
+L_SHAPE_TEMPLATE = [['...O.',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..O..',
+                     '..OO.',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '.O...',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['.OO..',
+                     '..O..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+T_SHAPE_TEMPLATE = [['..O..',
+                     '.OOO.',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '..OO.',
+                     '..O..',
+                     '.....',
+                     '.....'],
+                    ['.OOO.',
+                     '..O..',
+                     '.....',
+                     '.....',
+                     '.....'],
+                    ['..O..',
+                     '.OO..',
+                     '..O..',
+                     '.....',
+                     '.....']]
+
+PIECES = {'S': S_SHAPE_TEMPLATE,
+          'Z': Z_SHAPE_TEMPLATE,
+          'J': J_SHAPE_TEMPLATE,
+          'L': L_SHAPE_TEMPLATE,
+          'I': I_SHAPE_TEMPLATE,
+          'O': O_SHAPE_TEMPLATE,
+          'T': T_SHAPE_TEMPLATE}
+
+
+class GameState:
+    def __init__(self):
+        global FPSCLOCK, DISPLAYSURF, BASICFONT, BIGFONT
+        pygame.init()
+        FPSCLOCK = pygame.time.Clock()
+        DISPLAYSURF = pygame.display.set_mode((WINDOWWIDTH, WINDOWHEIGHT))
+        BASICFONT = pygame.font.Font('freesansbold.ttf', 18)
+        BIGFONT = pygame.font.Font('freesansbold.ttf', 100)
+        pygame.display.set_caption('Tetromino')
+
+        # DEBUG
+        self.total_lines = 0
+
+        # setup variables for the start of the game
+        self.board = self.getBlankBoard()
+        self.lastMoveDownTime = time.time()
```

Comment:
What are `lastMoveDownTime`, `lastMoveSidewaysTime`, and `lastFallTime` used for? it seems, from a quick search, that they are unused.